src/main/java/org/humancellatlas/ingest/schemas/SchemaService.java

```diff
@@ -52,10 +52,20 @@
                             .collect(Collectors.toList());
     }
 
+    public Schema getLatestSchemaByEntityType(String highLevelEntity, String entityType) {
+        List<Schema> allLatestSchema = filterLatestSchemas(highLevelEntity).stream()
+                .filter(schema -> schema.getConcreteEntity().matches(entityType))
+                .collect(Collectors.toList());
+
+        return allLatestSchema.size() > 0 ? allLatestSchema.get(0) : null;
+    }
+
     @Scheduled(fixedDelay = EVERY_24_HOURS)
     public void updateSchemasCollection() {
         String schemaBaseUri = environment.getProperty("SCHEMA_BASE_URI");
 
+        if (schemaBaseUri == null) return;
```

Comment:
I think this env var must always be set. You could raise an error here instead.

Reply:
I was not sure if I should throw an exception or just silently exit from this scheduled execution.
OK, I am going to throw an exception. Tx for the comment.

Reply:
I changed it to throw an exception and also added a test case for it.

Reply:
cool, thanks!!